src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -84,94 +81,61 @@ public ComplaintResource(AuthorizationCheckService authCheckService, ExerciseRep
+            }
+            authCheckService.isOwnerOfParticipationElseThrow((StudentParticipation) result.getParticipation());
+            savedComplaint = complaintService.createComplaint(complaint, result, complaint.examId(), principal);
         }
-
-        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.STUDENT, result.getParticipation().getExercise(), null);
-
-        // To build correct creation alert on the front-end we must check which type is the complaint to apply correct i18n key.
-        String entityName = complaint.getComplaintType() == ComplaintType.MORE_FEEDBACK ? MORE_FEEDBACK_ENTITY_NAME : COMPLAINT_ENTITY_NAME;
-        Complaint savedComplaint = complaintService.createComplaint(complaint, OptionalLong.empty(), principal);
-
-        // Remove assessor information from client request
-        savedComplaint.getResult().filterSensitiveInformation();
-
-        return ResponseEntity.created(new URI("/api/complaints/" + savedComplaint.getId()))
-                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, entityName, savedComplaint.getId().toString())).body(savedComplaint);
-    }
-
-    /**
-     * POST complaints/exam/examId: create a new complaint for an exam exercise
-     *
-     * @param complaint the complaint to create
-     * @param principal that wants to complain
-     * @param examId    the examId of the exam which contains the exercise
-     * @return the ResponseEntity with status 201 (Created) and with body the new complaints
-     * @throws URISyntaxException if the Location URI syntax is incorrect
-     */
-    @PostMapping("complaints/exam/{examId}") // TODO: should be exams/{examId}/(participations/{participationId}/)complaints
-    @EnforceAtLeastStudent
-    public ResponseEntity<Complaint> createComplaintForExamExercise(@PathVariable Long examId, @RequestBody Complaint complaint, Principal principal) throws URISyntaxException {
-        log.debug("REST request to save Complaint for exam exercise: {}", complaint);
-
-        validateNewComplaint(complaint);
-
-        Result result = resultRepository.findByIdElseThrow(complaint.getResult().getId());
-
-        // For non-exam exercises, the POST complaints should be used
-        if (!result.getParticipation().getExercise().isExamExercise()) {
-            throw new BadRequestAlertException("A complaint for an course exercise cannot be filed using this component", COMPLAINT_ENTITY_NAME,
-                    "complaintAboutCourseExerciseWrongComponent");
+        else {
+            if (result.getParticipation().getExercise().isExamExercise()) {
+                throw new BadRequestAlertException("A complaint for an exam exercise cannot be filed using this component", COMPLAINT_ENTITY_NAME,
+                        "complaintAboutExamExerciseWrongComponent");
+            }
+            authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.STUDENT, result.getParticipation().getExercise(), null);
```

Comment:
I see now. This is okay because we know that there is a participation for this exercise and for every exam exercise a user has access to that has a participation of that user, that user is allowed to complain to that participation.

Could you please add a comment to this so other people don't get confused about this difference in authorization-checks as I was?

Reply:
Added